PancakeSwap-Bots/Large-Delegators-Vote-Change/src/agent.spec.ts

```diff
@@ -0,0 +1,231 @@
+import { FindingSeverity, Finding, HandleTransaction, ethers } from "forta-agent";
+import { createAddress, TestTransactionEvent } from "forta-agent-tools/lib/tests";
+import { DELEGATE_VOTES_CHANGED_EVENT } from "./abi";
+import { NetworkManager } from "forta-agent-tools";
+import { NetworkData } from "./config";
+import bot from "./agent";
+import { createFinding } from "./findings";
+import { DECIMALS } from "./thresholds";
+import { BigNumber } from "ethers";
+import BN from "bignumber.js";
+
+function getPercentage(previousBalance: BigNumber, newBalance: BigNumber): string {
+  let pbalance_BN = new BN(previousBalance.toString());
+  let nbalance_BN = new BN(newBalance.toString());
+
+  return nbalance_BN.minus(pbalance_BN).dividedBy(pbalance_BN).multipliedBy(100).toString() + " %";
+}
+
+describe("delegate votes change bot", () => {
+  const MOCK_CONTRACT_ADDRESS = createAddress("0x1234");
+
+  let handleTransaction: HandleTransaction;
+  let eventInterface: ethers.utils.Interface;
+
+  let mockTxEvent: TestTransactionEvent;
+
+  let networkManager: NetworkManager<NetworkData>;
+
+  beforeEach(() => {
+    //create test transaction before each test
+    mockTxEvent = new TestTransactionEvent();
+  });
+
+  beforeAll(() => {
+    BN.set({ DECIMAL_PLACES: 5 });
+    const mockData: Record<number, NetworkData> = {
+      1111: {
+        cakeAddress: MOCK_CONTRACT_ADDRESS,
+      },
+    };
+
+    //initialize network manager with mock network
+    networkManager = new NetworkManager(mockData, 1111);
+
+    handleTransaction = bot.provideHandleTransaction(networkManager);
+
+    eventInterface = new ethers.utils.Interface([DELEGATE_VOTES_CHANGED_EVENT, "event MockEvent(uint)"]);
+  });
+
+  describe("handleTransaction", () => {
+    it("returns empty findings if no events are emitted,", async () => {
+      mockTxEvent.addAnonymousEventLog(MOCK_CONTRACT_ADDRESS, "", ...[]);
```

Comment:
this is fine but you can consider using `addInterfaceEventLog` for all addAnonymousEventLogs.